docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
+  
+
+#### **Alternative Considerations**
+
+1. **Encapsulate Score Calculation**:
+   - Alternative: Create separate classes for `ScoreCalculator`.
+      - Pros: Increases code modularity and allows for more scoring logic.
+      - Cons: Adds complexity that may not be necessary if the scoring logic remain simple.
+
+#### UML Class Diagram
+Below is the class diagram for the `QuizManager` class, illustrating its attributes and methods.
+![image](https://github.com/user-attachments/assets/85b30191-76ab-431f-8cdb-0e29c481e506)
+
+
+### True/False Feature Implementation
+
+The `TrueFalse` class is responsible for handling true/false questions within the quiz application. This section describes its design, implementation details, and the rationale behind design choices.
+
+#### Feature Overview
+The `TrueFalse` class represents a true/false question. It stores the question text and the correct answer as a boolean. Additionally, it includes methods to validate user answers and display answer options.
+
+#### Implementation Details
+
+1. **Attributes**:
+   - `boolean correctAnswer`: Stores the correct answer for the question.
+
+2. **Constructor**:
+   - `TrueFalse(String questionText, boolean correctAnswer)`:
+     Initializes the question text and the correct answer. The constructor checks that `questionText` is non-null and non-empty, ensuring a valid question is always provided.
+
+3. **Methods**:
+   - `checkAnswer(String userAnswer)`:
+     This method converts the user's input (`"true"` or `"false"`) into a boolean and compares it to the correct answer. It throws an `IllegalArgumentException` for invalid inputs, ensuring only `"true"` or `"false"` strings are accepted.
+   - `printOptions()`:
+     Displays answer choices ("1. True" and "2. False") for the user. This makes it clear what options are available for true/false questions.
+   - `toString()`:
+     Returns a string representation of the question with a "(True/False)" label to indicate its type.
+
+#### Design Rationale
 
+- **Boolean Storage for Correct Answer**: Storing the answer as a boolean simplifies the process of validating the user’s response since the comparison is a straightforward boolean check.
+- **Input Validation in `checkAnswer`**: This method ensures only `"true"` or `"false"` strings are accepted as valid answers, which prevents user input errors.
+- **Assertion Usage**: Assertions check that the question text and user answer are valid. This adds an additional layer of error handling during development.
+
+#### Alternative Considerations
+
+An alternative approach considered was to store `correctAnswer` as a `String` (`"true"` or `"false"`), which would simplify displaying it as text but would complicate validation. Using `boolean` was chosen for its simplicity and efficiency in logic checks.
+
+#### UML Class Diagram
+
+![TrueFalse Class Diagram](https://github.com/Lucky-Yuan00/tp/blob/jinlin-dev/docs/UML/TrueFalseClassDiagram.png?raw=true)
```

Comment:
Can check if the notation is in line with what we use in this course?
C circle is not needed and +/- signs should be used for public/private attributes or methods.